statsmodels/tsa/statespace/mlemodel.py

```diff
@@ -1741,25 +1742,28 @@ def bic(self):
         #         self.df_model * np.log(self.nobs_effective))
         return bic(self.llf, self.nobs_effective, self.df_model)
 
+
+    def _update_rank(self, singular_values):
+        self.model.update(self.params)
+        if self._rank is None:
+            self._rank = np.linalg.matrix_rank(np.diag(singular_values))
+        return
+
```

Comment:
I'm not sure this should be its own method? The `update` call technically is unrelated to the rank update, and the one liner `self._update_rank` isn't much different than the two liner `if self._rank is None: ...`

Reply:
Fair enough.  I think I added it somewhere along the line because I was annoyed at a few un-tested lines in the coverage results.  I'll revert it.